components/script/dom/canvasrenderingcontext2d.rs

```diff
@@ -31,14 +31,32 @@ use ipc_channel::ipc::IpcSender;
 use servo_url::ServoUrl;
 use std::mem;
 
+#[derive(JSTraceable, MallocSizeOf)]
+#[unrooted_must_root_lint::must_root]
+struct CanvasRenderingContext2DDroppableFields {
+    canvas_state: CanvasState,
+}
```

Comment:
An easier way to make this change would be to leave CanvasState as-is, and make CanvasRenderingContext2DDroppableFields only store the IpcSender that gets used in the destructor. We can call `canvas_state.get_ipc_renderer()` right after creating the `CanvasState` to obtain it.

Reply:
Yeah, that makes sense!

Reply:
I'm having trouble specifying the lifetime of the `ipc_renderer: &IpcSender<CanvasMsg>` because get_ipc_renderer() returns a reference. How do I tell rust to trust me and that reference will live long enough to be valid?

Reply:
I suggest cloning the sender and storing that instead.